src/analyze/analysis_handler.go

```diff
@@ -39,55 +56,62 @@ func AnalyzeBGPMessages(window common.Window, config *config.Configuration) {
 	fmt.Printf("ShakeAlert Outliers: \n%+v\n", shakeAlertOutliers)
 
 	//check the amount of the frequencies in results
-	if len(AllResults.AllFreq)+1 > maxPoints { //adding one more frequency would result in more than needed points
+	if (len(currResult.AllFreq) + 1) > maxPoints { //adding one more frequency would result in more than needed points
 		//remove the first item in the freq map; no modifications to the outlier map
 		firstResultKey := getSmallestTimestamp(getListOfKeys(frequencies))
-		delete(frequencies, *firstResultKey)
+		delete(frequencies, firstResultKey)
 
 		//now update the results
-		for timestamp := range lengthMap {
+		for timestamp, _ := range lengthMap {
 			_, resultVal := frequencies[timestamp]
-			if resultVal && frequencies[timestamp] < lengthMap[timestamp] { //if there exists a value at a given time stamp in the final results
-				frequencies[timestamp] = lengthMap[timestamp]
+			if resultVal { //if there exists a value at a given time stamp in the final results
+				if frequencies[timestamp] < lengthMap[timestamp] { //check if the existing value is smaller than the incoming
```

Comment:
this might be a daft question (I haven't fully parsed the whole function): is it possible for existing value to be greater than the incoming value at a timestamp? counts only ever increase, right?